### PR TITLE
Fix generation of dynamic filter

### PR DESCRIPTION
**Pull request by kewang1024**
1. Only create DynamicFilters when turned on to avoid unnecessary compute
2. Add dynamicFilterEnabled check

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java [hunk 1]
```diff
@@ -602,7 +608,7 @@ private static DynamicFiltersResult createDynamicFilters(
         {
             Map<String, VariableReferenceExpression> dynamicFilters = ImmutableMap.of();
             List<RowExpression> predicates = ImmutableList.of();
-            if (node.getType() == INNER && isEnableDynamicFiltering(session)) {
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java:605]: Why removing this?

Comment (kewang1024) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java:605]: This is inside the `createDynamicFilters` function, not in `visitJoin`

**Comment by kewang1024** [comment]
> Do we really need this patch? The operation is pretty cheap

Because currently we have a bug.
When we turn off dynamic filtering, we will still calculate the dynamic filter and then when we compare `!dynamicFilters.equals(node.getDynamicFilters()) ||`, it would be true, which is not expected. 
Thus we have to fix it in this PR

**Comment by highker** [comment]
OK, that makes sense. Is it ready for review? BTW, you don't need to make every PR a draft. Just publish it.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java [hunk 2]
```diff
@@ -602,7 +608,7 @@ private static DynamicFiltersResult createDynamicFilters(
         {
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java:608]: There is session in the param. Remove

**Pull request merged**